src/Service/LegacyTranslator.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ *NOTICE OF LICENSE
+ *
+ *This source file is subject to the Open Software License (OSL 3.0)
+ *that is bundled with this package in the file LICENSE.txt.
+ *It is also available through the world-wide-web at this URL:
+ *http://opensource.org/licenses/osl-3.0.php
+ *If you did not receive a copy of the license and are unable to
+ *obtain it through the world-wide-web, please send an email
+ *to license@prestashop.com so we can send you a copy immediately.
+ *
+ *DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ *versions in the future. If you wish to customize PrestaShop for your
+ *needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author INVERTUS UAB www.invertus.eu  <support@invertus.eu>
+ * @copyright SIX Payment Services
+ * @license   SIX Payment Services
+ */
+
+namespace Invertus\SaferPay\Service;
+
+use Invertus\SaferPay\Config\SaferPayConfig;
+use Invertus\SaferPay\Factory\ModuleFactory;
+
+class LegacyTranslator implements TranslatorInterface
+{
+    private const FILE_NAME = 'LegacyTranslator';
+
+    private $module;
+
+    public function __construct(ModuleFactory $moduleFactory)
+    {
+        $this->module = $moduleFactory->getModule();
+    }
+
+    public function translate(string $key): string
+    {
+        return $this->getTranslations()[$key] ?? $key;
+    }
+
+    private function getTranslations(): array
+    {
+        return [
+            SaferPayConfig::PAYMENT_ALIPAY => $this->module->l('Pay by Ali',  self::FILE_NAME),
```

Comment:
Pretty sure Saferpay only asked to translate crypto payment name, other should not be touched unless got the permission from client.